CMakeLists.txt

```diff
@@ -0,0 +1,1595 @@
+  )
+  add_dependencies(mixxx-lib libebur128)
+  target_include_directories(mixxx-lib PUBLIC
+    "${CMAKE_CURRENT_SOURCE_DIR}/lib/libebur128/ebur128"
+  )
+  target_link_libraries(mixxx-lib PUBLIC "${CMAKE_CURRENT_BINARY_DIR}/lib/libebur128/${CMAKE_STATIC_LIBRARY_PREFIX}ebur128${CMAKE_STATIC_LIBRARY_SUFFIX}")
+else()
+  message(STATUS "Linking libebur128 dynamically")
+  target_link_libraries(mixxx-lib PUBLIC Ebur128::Ebur128)
+endif()
+
+# FidLib
+add_library(fidlib OBJECT lib/fidlib/fidlib.c)
+if(MSVC)
+  target_compile_definitions(fidlib PRIVATE T_MSVC)
+elseif(MINGW)
+  target_compile_definitions(fidlib PRIVATE T_MINGW)
+else()
+  target_compile_definitions(fidlib PRIVATE T_LINUX)
+endif()
+target_include_directories(mixxx-lib PUBLIC lib/fidlib)
+target_link_libraries(mixxx-lib PUBLIC fidlib)
+
+# FLAC
+find_package(FLAC REQUIRED)
+target_link_libraries(mixxx-lib PUBLIC FLAC::FLAC)
+
+# FpClassify This is a wrapper around the fpclassify function that prevents
+# inlining It is compiled without optimization and allows to use these function
+# from -ffast-math optimized objects
+add_library(FpClassify STATIC src/util/fpclassify.cpp)
+target_link_libraries(mixxx-lib PUBLIC FpClassify)
+
+# gtest
+target_include_directories(mixxx-lib PUBLIC lib/gtest-1.7.0/include)
+
+# LAME
+find_package(LAME REQUIRED)
+target_link_libraries(mixxx-lib PUBLIC LAME::LAME)
+
+# OpenGL
+set(OpenGL_GL_PREFERENCE "GLVND")
+find_package(OpenGL REQUIRED)
+target_link_libraries(mixxx-lib PUBLIC OpenGL::GL)
+
+# Ogg Vorbis
+find_package(OggVorbis REQUIRED)
+target_include_directories(mixxx-lib PUBLIC ${OggVorbis_INCLUDE_DIRS})
+target_link_libraries(mixxx-lib PUBLIC ${OggVorbis_LIBRARIES})
+
+# PortAudio
+find_package(PortAudio REQUIRED)
+target_include_directories(mixxx-lib PUBLIC ${PORTAUDIO_INCLUDE_DIRS})
+target_link_libraries(mixxx-lib PUBLIC ${PORTAUDIO_LIBRARIES})
+
+# PortAudio Ring Buffer
+add_library(PortAudioRingBuffer STATIC lib/portaudio/pa_ringbuffer.c)
+target_include_directories(mixxx-lib PUBLIC lib/portaudio)
+target_link_libraries(mixxx-lib PUBLIC PortAudioRingBuffer)
+
+# PortMidi
+find_package(PortMidi REQUIRED)
+target_include_directories(mixxx-lib PUBLIC ${PortMidi_INCLUDE_DIRS})
+target_link_libraries(mixxx-lib PUBLIC ${PortMidi_LIBRARIES})
+
+# Protobuf
+add_subdirectory(src/proto)
+target_link_libraries(mixxx-lib PUBLIC mixxx-proto)
+
+# Qt
+find_package(Qt5
+  COMPONENTS
+    Concurrent
+    Core
+    Gui
+    Network
+    OpenGL
+    Script
+    ScriptTools
+    Sql
+    Svg
+    Test
+    Widgets
+    Xml
+  REQUIRED
+)
+target_link_libraries(mixxx-lib PUBLIC
+  Qt5::Concurrent
+  Qt5::Core
+  Qt5::Gui
+  Qt5::Network
+  Qt5::OpenGL
+  Qt5::Script
+  Qt5::ScriptTools
+  Qt5::Sql
+  Qt5::Svg
+  Qt5::Test
+  Qt5::Widgets
+  Qt5::Xml)
+if(UNIX AND NOT APPLE)
+  find_package(X11 REQUIRED)
+  find_package(Qt5 COMPONENTS X11Extras DBus REQUIRED)
+  target_link_libraries(mixxx-lib PUBLIC X11 Qt5::X11Extras Qt5::DBus)
```

Comment:
This does not work for me because of the hardcoded `X11`.
I think it should be like this:
`target_link_libraries(mixxx-lib PUBLIC X11::X11 Qt5::X11Extras Qt5::DBus)`

Reply:
Fixed via https://github.com/mixxxdj/mixxx/pull/2280/commits/f28e531228bc11c794471611f4b7cc8d86e82db0